app-backend/common/src/main/scala/Authentication.scala

```diff
@@ -32,82 +37,40 @@ trait Authentication extends Directives {
     * Authenticates user based on bearer token (JWT)
     */
   def authenticate: Directive1[User] = {
-    validateTokenHeader.flatMap { validToken =>
-      JwtJson4s.decodeJson(validToken, auth0Secret, Seq(JwtAlgorithm.HS256)) match {
-        case Success(parts) =>
-          val sub = (parts \ "sub").extract[String]
-
-          onSuccess(Users.getUserById(sub)).flatMap {
-            case Some(user) => provide(user)
-            case None => onSuccess(Users.createUserWithAuthId(sub)).flatMap {
-              user => provide(user)
-            }
-          }
-        case Failure(_) => reject(AuthenticationFailedRejection(CredentialsRejected, challenge))
-      }
+    extractTokenHeader.flatMap { token =>
+      authenticateWithToken(token)
     }
   }
 
   /**
     * Authenticates user based on bearer token (JWT)
     */
   def authenticateWithParameter: Directive1[User] = {
-    validateTokenParameter.flatMap { validToken =>
-      getUserWithJWT(validToken)
+    parameter('token).flatMap { token =>
+      authenticateWithToken(token)
     }
   }
 
-  def authenticateWithToken(token: String): Directive1[User] = {
-    if (Jwt.isValid(token, auth0Secret, Seq(JwtAlgorithm.HS256))) {
-      getUserWithJWT(token)
-    } else {
-      reject(AuthenticationFailedRejection(CredentialsRejected, challenge))
-    }
+  def verifyJWT(tokenString: String): Either[BadJWTException, (JwtToken, JWTClaimsSet)] = {
+    val token: JwtToken = JwtToken(content = tokenString)
+    val url = "https://raster-foundry-dev.auth0.com/.well-known/jwks.json"
```

Comment:
@hectcastro This is the URL that needs to be an environment variable. We can probably name the environment variable something not Auth0 specific like `JWKS_URL`.

Reply:
Other Auth0 environment variables can likely be removed